src/cacheUntilType.js

```diff
@@ -0,0 +1,51 @@
+import { OuterSubscriber } from 'rxjs/OuterSubscriber';
+import { subscribeToResult } from 'rxjs/util/subscribeToResult';
+
+
+export function cacheUntilType(type) {
+  return this.lift(new CacheUntilTypeOperator(type));
+}
+
+class CacheUntilTypeOperator {
```

Comment:
Sure. Went with classes original as it seemed to be the pattern established by RxJSv5 source but I'm not attached.